src/tools/rbd_nbd/rbd-nbd.cc

```diff
@@ -466,6 +471,38 @@ static int open_device(const char* path, bool try_load_moudle = false)
   return nbd;
 }
 
+static bool find_key_from_conf_file(const char *key)
+{
+  std::vector <std::string> my_sections;
+  g_conf->get_my_sections(my_sections);
+  std::string admin_socket_str;
+  return ((g_conf->get_val_from_conf_file(my_sections, key,
+           admin_socket_str, false) == 0) ? true : false);
```

Comment:
Actually, trying your version, it looks like it works like expected: when admin-socket is specified either in config or via command line, this setting is used, otherwise it is set to {run_dir}/ceph-client.{nbd_dev}.asok.

I have many comments to your code though, but at first I think we need to reach a consensus on the default behavior (Jason wants asok is not set at all by default, and this is actually the safest behavior, though may be not the most useful).

@dillaman What do you think?

Looking at how asok could be used to improve `nbd list` command, I tend to agree with the idea to hard code it to something like {run_dir}/rbd-nbd.{nbd_dev}.asok (so user settings are ignored). Then `ndb list` could go through the list of rbd-nbd.*.asok files, and query through asok commands. Also, this could also allow to unmap using image name, not nbd device.